src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -14,11 +14,13 @@
  * limitations under the License.
  */
 
+#include <algorithm>
```

Comment:
This is known to be extremely heavy and we should avoid it if possible. What APIs do we need from it here?